src/java.compiler/share/classes/javax/lang/model/util/Elements.java

```diff
@@ -758,6 +758,29 @@ default boolean isAutomaticModule(ModuleElement module) {
         return false;
     }
 
+    /**
+     * {@return the class body of an {@code enum} constant if the
+     * argument is an {@code enum} constant declared with an optional
+     * class body, {@code null} otherwise}
+     *
+     * @implSpec
+     * The default implementation of this method throws {@code
+     * UnsupportedOperationException} if the argument is an {@code
+     * enum} constant and throws an {@code IllegalArgumentException}
+     * if it is not.
+     *
+     * @param enumConstant an enum constant
+     * @throws IllegalArgumentException if the argument is not an {@code enum} constant
+     * @jls 8.9.1 Enum Constants
+     * @since 22
+     */
+    default TypeElement getEnumConstantBody(VariableElement enumConstant) {
+        switch(enumConstant.getKind()) {
+        case ENUM_CONSTANT -> throw new UnsupportedOperationException();
```

Comment:
Hm... Most retrofitted methods in this API return null, not throw UnsupportedOperationException. The only other exception seems to be Elements.getFileObjectOf. Is this the new approach, or it just does make more sense for these two particular methods?

Reply:
I think it depends whether a `null` result is permitted by the specification of the method.  For this method, while `null` would be the correct result most of the time, it is exactly the wrong result when it is most interesting, and so a non-exception result would be misleading.

Reply:
I see your point and agree with it: the `default` implementation will return `null` if called for an enum constant with the class body. That said, isn't, say, `Elements.getTypeElement(ModuleElement, CharSequence)` in a similar situation? But it returns `null`.